File: CS/ComplexDialogSample.Module/BusinessObjects/BusinessObjects.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using System;
using System.ComponentModel;

namespace ComplexDialogSample.Module.BusinessObjects {

    [NavigationItem("Resources")]
    public class Team : BaseObject {
        public Team(Session session) : base(session) { }
        public string Name { get { return GetPropertyValue<string>("Name"); } set { SetPropertyValue<string>("Name", value); } }
    }

    [NavigationItem("Process")]
    [DefaultProperty("Index")]
    public class Office : BaseObject {
        public Office(Session session) : base(session) { }
        public string Index { get { return GetPropertyValue<string>("Index"); } set { SetPropertyValue<string>("Index", value); } }
        public string OccupiedBy { get { return GetPropertyValue<string>("OccupiedBy"); } set { SetPropertyValue<string>("OccupiedBy", value); } }
    }

    [NavigationItem("Process")]
    public class Order : BaseObject {
        public Order(Session session) : base(session) { }
        public Team Team { get { return GetPropertyValue<Team>("Team"); } set { SetPropertyValue<Team>("Team", value); } }
        public Office Office { get { return GetPropertyValue<Office>("Office"); } set { SetPropertyValue<Office>("Office", value); } }
        public Service Service { get { return GetPropertyValue<Service>("Service"); } set { SetPropertyValue<Service>("Service", value); } }
        public DateTime DueDate { get { return GetPropertyValue<DateTime>("DueDate"); } set { SetPropertyValue<DateTime>("DueDate", value); } }
    }

    [NavigationItem("Resources")]
    [DefaultProperty("Description")]
    public class Service : BaseObject {
        public Service(Session session) : base(session) { }
        public string Description { get { return GetPropertyValue<string>("Description"); } set { SetPropertyValue<string>("Description", value); } }
    }
}

File: VB/ComplexDialogSample.Module/BusinessObjects/BusinessObjects.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports System.ComponentModel

Namespace ComplexDialogSample.Module.BusinessObjects

    <NavigationItem("Resources")>
    Public Class Team
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Name As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", value)
            End Set
        End Property
    End Class

    <NavigationItem("Process")>
    <DefaultProperty("Index")>
    Public Class Office
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Index As String
            Get
                Return GetPropertyValue(Of String)("Index")
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Index", value)
            End Set
        End Property

        Public Property OccupiedBy As String
            Get
                Return GetPropertyValue(Of String)("OccupiedBy")
            End Get

            Set(ByVal value As String)
                SetPropertyValue("OccupiedBy", value)
            End Set
        End Property
    End Class

    <NavigationItem("Process")>
    Public Class Order
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Team As Team
            Get
                Return GetPropertyValue(Of Team)("Team")
            End Get

            Set(ByVal value As Team)
                SetPropertyValue("Team", value)
            End Set
        End Property

        Public Property Office As Office
            Get
                Return GetPropertyValue(Of Office)("Office")
            End Get

            Set(ByVal value As Office)
                SetPropertyValue("Office", value)
            End Set
        End Property

        Public Property Service As Service
            Get
                Return GetPropertyValue(Of Service)("Service")
            End Get

            Set(ByVal value As Service)
                SetPropertyValue("Service", value)
            End Set
        End Property

        Public Property DueDate As Date
            Get
                Return GetPropertyValue(Of Date)("DueDate")
            End Get

            Set(ByVal value As Date)
                SetPropertyValue("DueDate", value)
            End Set
        End Property
    End Class

    <NavigationItem("Resources")>
    <DefaultProperty("Description")>
    Public Class Service
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Description As String
            Get
                Return GetPropertyValue(Of String)("Description")
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Description", value)
            End Set
        End Property
    End Class
End Namespace
